src/blocks/github.rs

```diff
@@ -0,0 +1,277 @@
+        let token = match std::env::var(GITHUB_TOKEN_ENV).ok() {
+            Some(v) => v,
+            None => {
+                return Err(BlockError(
+                    "github".to_owned(),
+                    "missing GITHUB_TOKEN environment variable".to_owned(),
+                ))
+            }
+        };
+
+        Ok(Github {
+            id: Uuid::new_v4().to_simple().to_string(),
+            update_interval: block_config.interval,
+            text: TextWidget::new(config).with_text("x").with_icon("github"),
+            api_server: block_config.api_server,
+            token,
+            format: FormatTemplate::from_string(&block_config.format)
+                .block_error("github", "Invalid format specified")?,
+        })
+    }
+}
+
+impl Block for Github {
+    fn update(&mut self) -> Result<Option<Update>> {
+        let aggregations = match Notifications::new(&self.api_server, &self.token).try_fold(
+            map!("total".to_owned() => 0),
+            |mut acc,
+             notif|
+             -> std::result::Result<HashMap<String, u64>, Box<dyn std::error::Error>> {
+                let n = notif?;
+                acc.entry(n.reason).and_modify(|v| *v += 1).or_insert(1);
+                acc.entry("total".to_owned()).and_modify(|v| *v += 1);
+                Ok(acc)
+            },
+        ) {
+            Ok(v) => v,
+            Err(_) => {
+                // If there is a error reported, set the value to x
+                self.text.set_text("x".to_owned());
+                return Ok(Some(self.update_interval.into()));
+            }
+        };
+
+        let default: u64 = 0;
+        let values = map!(
+            "{total}" => format!("{}", aggregations.get("total").unwrap_or(&default)),
+            // As specified by:
+            // https://developer.github.com/v3/activity/notifications/#notification-reasons
+            "{assign}" => format!("{}", aggregations.get("assign").unwrap_or(&default)),
+            "{author}" => format!("{}", aggregations.get("author").unwrap_or(&default)),
+            "{comment}" => format!("{}", aggregations.get("comment").unwrap_or(&default)),
+            "{invitation}" => format!("{}", aggregations.get("invitation").unwrap_or(&default)),
+            "{manual}" => format!("{}", aggregations.get("manual").unwrap_or(&default)),
+            "{mention}" => format!("{}", aggregations.get("mention").unwrap_or(&default)),
+            "{review_requested}" => format!("{}", aggregations.get("review_requested").unwrap_or(&default)),
+            "{security_alert}" => format!("{}", aggregations.get("security_alert").unwrap_or(&default)),
+            "{state_change}" => format!("{}", aggregations.get("state_change").unwrap_or(&default)),
+            "{subscribed}" => format!("{}", aggregations.get("subscribed").unwrap_or(&default)),
+            "{team_mention}" => format!("{}", aggregations.get("team_mention").unwrap_or(&default))
```

Comment:
Since these are hardcoded here we might as well mention them in `blocks.md` to save the user having to go check the github api page.